docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,287 @@
+  "ctypes". The C/C++ scripts can exist inside your reader module, and are
+  compiled as part of the ctapipe installation. See
+  `ctapipe.utils.neighbour_sum` as an example of how to interface Python with
+  C code, and remember to add your extension to ext_modules in setup.py.
+* **Your reader exists as an external C/C++ library**, therefore software such
+  as SWIG can be used to provide a Python wrapper to the library,
+  automatically creating interfaces to the functions in the library.
+
+
+EventFileReaders
+================
+
+The module `ctapipe.io.unofficial.eventfilereader` contains the
+`ctapipe.io.eventfilereader.EventFileReader` classes for the unofficial
+readers. These classes provides the high-level interface for reading data
+from each data format. Read about the purpose of EventFileReaders at
+:ref:`EventFileReaderClasses`.
+
+Each reader is required to import their
+`Low-Level Reader-Specific Modules`_ inside their `__init__` method,
+therefore bypassing the need for the external reader software to be installed
+unless someone has chosen to use this specific
+`ctapipe.io.eventfilereader.EventFileReader`, in which case a
+`ModuleNotFoundError` will be raised if the external software has not been
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.is_compatible`. This
+method attempts to find a reader which *may* be compatible with the input
+file, and can be as simple as looking for a unique file extension in the file
+path or attempting to find a header keyword in the file. It is strongly
+advised that this method does not depend on the external reader software, as
+this would introduce a dependency when choosing the correct reader in the
+`ctapipe.io.eventfilereader.EventFileReaderFactory`.
+
+The currently existing EventFileReaders,
+`ctapipe.io.eventfilereader.HessioFileReader` and
+`ctapipe.io.unofficial.eventfilereader.TargetioFileReader` can be used as
+templates to create a new EventFileReader for a different data format.
+
+
+Low-Level Reader-Specific Modules
+===========================
+
+A seperate module should be created for the low-level method used for filling
+the event containers for a specific data format. The naming convention for
+this module is "\*io.py", where "\*" is the short-name for the data format.
+For example: `ctapipe.io.hessio` and `ctapipe.io.unofficial.targetio.targetio`.
+
+The low-level module can be a directory of multiple modules, allowing the
+various components involved in the event building (such as containers) to be
+grouped together for a particular data format.
+
+Each data format can be wildly different, therefore these low-level
+modules can appear very different in how they function. This is fine as long
+as these rules are followed:
+
+* The `ctapipe.io.eventfilereader.EventFileReader` calls the low-level methods
+  such that the high-level interface remains the same as other
+  `ctapipe.io.eventfilereader.EventFileReader`.
```

Comment:
As it is a Base class of `EventFileReader` this does not need to be a rule as it is enforced syntactically.